### DISCUSSION TITLE
Which Delphi versions does ZUGFeRD-for-Delphi support?

### DISCUSSION THREAD

**Discussion by Bodenseematze** [LandrixSoftware/ZUGFeRD-for-Delphi#32] [Q&A] [answered]
And how do I compile the library / package(s)?
And where is the developers documentation how to use the library / package in own projects?

All I found was documentation about the different file formats - but nothing about the structure of the delphi units/classes etc. and also nothing about the steps for integrating it in own applications...
Am I missing something or isn't there any information?

**Comment by landrix** [comment, marked as answer]
Currently this lib needs Delphi support for generics. Simple include the needed Units in your project, An example for reading can found here
https://github.com/LandrixSoftware/XRechnung-for-Delphi/blob/master/intf.XRechnung.pas

More infos are planed in november, when both libs are combined to a all-in-one solution.

**Comment by Bodenseematze** [comment]
Aah - that explains why I was not able to compile the sample...
...what does "currently" mean? Are there plans to support older versions?
I would need support for Delphi 7... ;-)

**Comment by landrix** [comment]
yes, support back to delphi 6 is planned

**Reply by Bodenseematze** [reply]
I hope this will be happen soon ;-)
